### DISCUSSION TITLE
OpenWeedLocator feedback

### DISCUSSION THREAD

**Discussion by geezacoleman** [geezacoleman/OpenWeedLocator#48] [General]
Presented the OWL at the Mingenew Expo on Friday last week and received some great feedback from growers and industry. Thought it would be a good idea to summarise it here, and start a thread for feedback on the units too if anyone has any other general comments.

![image](https://user-images.githubusercontent.com/51358498/184787770-acebab1a-07dd-4ce2-9abe-65816f18dd87.png)

**General Comments**
Overall comments were positive about accessibility, low-cost and open-source nature of the project. Interest from schools about the project as an educational tool.

**Design**
* absolutely needs to be as rugged and robust as possible. Features don't work on broken units. Design is likely to change to meet this - current units don't keep dust or chemical out to the level that is required.
  * if it is too rugged/tightly packed the educational benefit decreases, so need to be careful about maximising performance at the expense of educational benefits. There is potential for both educational and production versions potentially so people can choose.
* ease of assembly needs to be improved - the small wires going from the relay board to the Pi are difficult to assign to the correct GPIO pins - a relay hat will be part of future designs - my least favourite parts of assembly are the small wire connections!
* mounting - needs to be more user friendly and less fiddly. Perhaps a sleeve to fit inside standard dimension RHS steel for easy welding.

**Software**
* performance was quite good in the bright sunlight - very quick response to weeds
* communication between OWL units is a must - current method of multiple independent OWLs makes wider systems prohibitively complex to use and update.
* Green on green would be wonderful - BUT - no point having it if the above issues aren't managed.

**Planned Changes**
Based on these comments, the priority list of development for the system is then:
* Improve ease of assembly by minimising connections
* Increase robustness of the OWL
* Develop communication tool to manage multiple OWLs from a central system in the cab
* Green-on-green and other software features are great, but only once the hardware is useable.

What are everyone's thoughts on these points? Would be great to hear feedback from other users and developers of systems too.

**Comment by EchelonMirror** [comment]
Relay hat is the best idea for the GPIO. 

If you know anyone who has the tools to add rivets, then you can use RHS aluminium for the general structure of the boom, and only need to weld the 50mm steel for the hitch. [https://twitter.com/TRethus/status/1430045676000272385/photo/1](url). If you need the schematic I can provide it. 

If you need a communication protocol I recommend using MQTT, https://github.com/eclipse/mosquitto. Quick to get up and running on a rudimentary level. Unless you've got a P2P capable OS, you're going to need to have a 'core' unit that acts as a server - and on that you're going to need a wifi router like a NightHawk etc. Wifi is a lot easier that having ethernet cables running everywhere and having to be plugged into a switch.

You can make a case that is robust (IP67) and easily assembled. If you take a rated case from a notable supplier, and insert a 3D printed skeleton inside this will help with the ease of installing and cable routing.

**Comment by geezacoleman** [comment]
Draft design changes and files provided in #19

**Comment by geezacoleman** [comment]
Some good ideas from a few different field days recently:
1. these could be mounted between the front and the header to spot spray melons immediately post harvest. 
  a. harvesters travel slow anyway and melons would be large, so would be ideal conditions for the OWL
